New score proto

cghawthorne opened the pull request.

cghawthorne committed:
New score proto

czhuang commented on magenta/protobuf/music.proto:
```diff
@@ -136,4 +145,22 @@ message NoteSequence {
+    // We can expect perfect timing alignment with measures and complete
+    // TimeSignature and KeySignature information.
+    SCORE = 1;
+  }
```
this is a general music proto?  would we include other source types like midi?

cghawthorne replied:
Chatted in person. Will add new types as we find uses for them.

czhuang commented on magenta/protobuf/music.proto:
```diff
@@ -136,4 +145,22 @@ message NoteSequence {
   }
+
+  // Stores score-related information about a particular part (sometimes also
+  // called track or voice).
```
do we have to include which staff this track/voice was part of to inject notes back into musescore?

cghawthorne replied:
We can derive the staff by part / 4, and voice is part % 4.

czhuang commented on magenta/protobuf/score.proto:
```diff
@@ -0,0 +1,99 @@
+message Chord {
+  // The track index.
+  int32 track = 1;
+  // The duration of the chord.
```
[repeat] do we have to include which staff this track/voice was part of to inject notes back into musescore?

cghawthorne replied:
the track field should take care of that

czhuang commented on magenta/protobuf/score.proto:
```diff
@@ -0,0 +1,99 @@
+  // The measure-relative tick position of the staff element. For example if a
+  // chord is at the very beginning of the measure, this would be 0. If it's on
+  // the second beat of a 4/4 measure, it would be 960.
+  int32 measure_local_tick = 1;
```
alternative name, "beat_tick"?  and this is "start_time_beat_tick"?  and the notes could be of different duration?

cghawthorne replied:
Will change to local_beat_tick.

Update comment to say it's the start time.

czhuang replied:
or perhaps "onset_local_tick"? this makes it super clear it's start time and also since we're measuring by ticks, beats are derived and not directly encoded.

cghawthorne replied:
Done

czhuang commented on magenta/protobuf/score.proto:
```diff
@@ -0,0 +1,99 @@
+message Note {
+  // The midi pitch of the note.
+  int32 pitch = 1;
+}
```
include duration in ticks?

cghawthorne replied:
Duration is specified in the Chord.

cghawthorne commented on magenta/protobuf/score.proto:
```diff
@@ -0,0 +1,99 @@
+  string name = 2;
+}
+
+// tick values from: 
```
Move comments to reasonable place

cghawthorne committed:
Updated comments and reverted score-relative times

cghawthorne committed:
fix name

czhuang commented on magenta/protobuf/music.proto:
```diff
@@ -51,21 +51,30 @@ message NoteSequence {
     double end_time = 4;
-    // Score-relative note length. E.g. a quarter note is 1/4.
-    int32 numerator = 5;
-    int32 denominator = 6;
```
just a note for us to add this back in

cghawthorne replied:
Done

vug committed:
import StringIO in both Python 2 & 3 (#80)

Python 3 removed StringIO and cStringIO modules. Instead there is
io module. From https://docs.python.org/3.0/whatsnew/3.0.html
"The StringIO and cStringIO modules are gone. Instead,
import the io module and use io.StringIO or io.BytesIO for text
and data respectively."

cghawthorne committed:
Disable basic_rnn_train_test. (#84)

Should turn back on when the test doesn't time out.

cghawthorne committed:
change segment start time name

cghawthorne committed:
Merge branch 'master' into scoreproto

czhuang commented on magenta/protobuf/music.proto:
```diff
@@ -51,6 +51,12 @@ message NoteSequence {
+
   message Note {
     // MIDI pitch; see en.wikipedia.org/wiki/MIDI_Tuning_Standard for details.
     int32 pitch = 1;
```
i just remembered we wanted to add the pitch spellings from MuseScore?

cghawthorne replied:
good catch. i'll do that in a followup pr
